timely/src/dataflow/scopes/mod.rs

```diff
@@ -33,6 +33,10 @@ pub trait Scope: ScopeParent {
     /// A sequence of scope identifiers describing the path from the worker root to this scope.
     fn addr(&self) -> Vec<usize>;
 
+    /// A sequence of scope identifiers describing the path from the worker root to the index in
+    /// this scope.
+    fn addr_for(&self, index: usize) -> Vec<usize>;
```

Comment:
Oh boy. You did have the docs and I read the files out of order. I'm sorry! But maybe "to the index" -> "to the child indicated by index"?